src/Tickets/Commerce/Admin/Edit_Purchaser_Provider.php

```diff
@@ -0,0 +1,258 @@
+					if ( ! $sent ) {
+						wp_send_json_error(
+							_x(
+								'There was an error sending the email receipt. Please ensure WordPress is configured for email delivery.',
+								'When the purchaser email receipt fails for an unknown reason.',
+								'event-tickets'
+							)
+						);
+						return;
+					}
+				}
+
+				if ( $updated ) {
+					wp_send_json_success(
+						[
+							'name'  => $first_name . ' ' . $last_name,
+							'email' => $email,
+						]
+					);
+				} else {
+					wp_send_json_error(
+						_x(
+							'There was an unknown error while updating the purchaser. Please try again later.',
+							'When the purchaser update fails for an unknown reason.',
+							'event-tickets'
+						)
+					);
+				}
+				break;
+			case 'GET':
+				// Fetch the order.
+				$post  = get_post( sanitize_text_field( $_GET['ID'] ?? null ) );
+				$order = tec_tc_get_order( $post );
+
+				if ( ! $order ) {
+					wp_send_json_error(
+						_x(
+							'There was an unknown error while retrieving the purchaser. Please try again later.',
+							'When the purchaser GET request fails for an unknown reason.',
+							'event-tickets'
+						)
+					);
+					return;
+				}
+
+				wp_send_json_success( $order->purchaser );
+				break;
+		}
+	}
+
+	/**
+	 * Callback that renders the modal with context applied.
+	 *
+	 * @since TBD
+	 *
+	 * @param mixed           $html The html param from the filter.
+	 * @param string          $file The file path.
+	 * @param string          $name The name of the file.
+	 * @param Tribe__Template $template The template object.
+	 * @param mixed           $context The context data.
+	 *
+	 * @return mixed The passed html.
+	 */
+	public function render_modal( $html, $file, $name, Tribe__Template $template, $context ) {
+		$dialog_view    = tribe( 'dialog.view' );
+		$title          = esc_html_x( 'Edit purchaser', 'Edit purchaser modal title.', 'event-tickets' );
+		$admin_template = tribe( 'tickets.admin.views' );
+
+		ob_start();
+		$dialog_view->render_modal(
+			$admin_template->template(
+				'src/admin-views/commerce/orders/single/edit-purchaser-modal',
+				$context,
+				false
+			),
+			[
+				'id'             => 'edit-purchaser-modal',
+				'append_target'  => '#edit-purchaser-modal-container',
+				'button_display' => false,
+				'title'          => $title,
+				'close_event'    => 'tecTicketsCommerceClosePurchaserModal',
+				'show_event'     => 'tecTicketsCommerceOpenPurchaserModal',
+			],
+			'edit-purchaser-modal'
+		);
+
+		$modal_content = ob_get_clean();
+		$modal         = '<div class="tribe-common"><div id="edit-purchaser-modal-container"></div></div><div>';
+		$modal        .= $modal_content;
+		$modal        .= '</div>';
+
+		// phpcs:ignore StellarWP.XSS.EscapeOutput.OutputNotEscaped, WordPress.Security.EscapeOutput.OutputNotEscaped
+		echo $modal;
+
+		// Always return $html - we are not overriding the template, simply "appending" our modal.
+		return $html;
```

Comment:
A couple of more questions about this (sorry):
1. Since we're filtering a template, what happens when the template is set to be returned and not rendered (echoed)?
2. Would appending be `return $html . $modal;`?

Reply:
This filter is an override. If I return ANY result it will short-circuit the template being loaded. So in order to "append" I output directly, and leave the `$html` var alone so it continues rendering the template.